libyul/AssemblyStack.h

```diff
@@ -69,12 +70,24 @@ class AssemblyStack: public langutil::CharStreamProvider
 	enum class Machine { EVM, Ewasm };
 
 	AssemblyStack():
-		AssemblyStack(langutil::EVMVersion{}, Language::Assembly, solidity::frontend::OptimiserSettings::none())
+		AssemblyStack(
+			langutil::EVMVersion{},
+			Language::Assembly,
+			solidity::frontend::OptimiserSettings::none(),
+			langutil::DebugInfoSelection::Default()
+		)
 	{}
-	AssemblyStack(langutil::EVMVersion _evmVersion, Language _language, solidity::frontend::OptimiserSettings _optimiserSettings):
+
+	AssemblyStack(
+		langutil::EVMVersion _evmVersion,
+		Language _language,
+		solidity::frontend::OptimiserSettings _optimiserSettings,
+		langutil::DebugInfoSelection const& _debugInfoSelection
```

Comment:
I could avoid having to change all the places that create `AssemblyStack` by giving this a default but it also makes it very easy to forget to pass it in, which could bypass the CLI/JSON setting.